WalletWasabi/WebClients/PayJoin/PayjoinClient.cs

```diff
@@ -69,8 +76,21 @@ public async Task<PSBT> RequestPayjoin(PSBT originalTx, IHDKey accountKey, Roote
 
 			cloned.GlobalXPubs.Clear();
 
-			var bpuResponse = await HttpClient.SendAsync(HttpMethod.Post, "",
-				new StringContent(cloned.ToHex(), Encoding.UTF8, "text/plain"), cancellationToken).ConfigureAwait(false);
+			var request = new HttpRequestMessage(HttpMethod.Post, PaymentUrl)
+			{
+				Content = new StringContent(cloned.ToHex(), Encoding.UTF8, "text/plain")
+			};
+
+			if (TorHttpClient is null && ClearnetHttpClient is null)
+			{
+				throw new NullReferenceException(
+					"A Tor HttpClient OR a clearnet HttpClient needs to be provided to the Payjoin client");
+			}
+
+			HttpResponseMessage bpuResponse = TorHttpClient is { } ?
+				await TorHttpClient.SendAsync(request, cancellationToken).ConfigureAwait(false) :
+				await ClearnetHttpClient.SendAsync(request, cancellationToken).ConfigureAwait(false);
```

Comment:
This way we don't check `TorHttpClient` two times.
Also it might fix CodeScene.

```suggestion
			HttpResponseMessage bpuResponse = null;
			if (TorHttpClient is null)
			{
				if (ClearnetHttpClient is null)
				{
					throw new NullReferenceException(
						"A Tor HttpClient OR a clearnet HttpClient needs to be provided to the Payjoin client");
				}
				else
				{
					bpuResponse = await ClearnetHttpClient.SendAsync(request, cancellationToken).ConfigureAwait(false);
				}
			}
			else
			{
				bpuResponse = await TorHttpClient.SendAsync(request, cancellationToken).ConfigureAwait(false);
			}
```